presto-main/src/main/java/io/prestosql/dispatcher/DispatchInfo.java

```diff
@@ -52,21 +63,41 @@ private DispatchInfo(Optional<CoordinatorLocation> coordinatorLocation, Optional
         this.queuedTime = requireNonNull(queuedTime, "queuedTime is null");
     }
 
+    /**
+     * Returns the coordinator's uri.
+     *
+     * @return an Option {@link CoordinatorLocation}.
+     */
     public Optional<CoordinatorLocation> getCoordinatorLocation()
     {
         return coordinatorLocation;
     }
 
+    /**
+     * Returns if there was a failure during the query.
+     *
+     * @return an Optional {@link ExecutionFailureInfo}.
+     */
     public Optional<ExecutionFailureInfo> getFailureInfo()
     {
         return failureInfo;
     }
 
+    /**
+     * Returns the elapsed time of the query.
+     *
+     * @return a {@link Duration} object.
+     */
     public Duration getElapsedTime()
     {
         return elapsedTime;
     }
 
+    /**
+     * Returns the waiting time for a query.
```

Comment:
returns the queued time. The waiting time does not make it clear that the query hasn't started yet.